### PR TITLE
[Utils] Add clang tool in manylinux2014_x86_64 docker image

### PR DESCRIPTION
**Pull request by Officeyutong**
Also see https://github.com/WasmEdge/WasmEdge/pull/2570#issuecomment-1595134834

This pull request adds installing process of clang when building the image. Some of tests require clang to build intermediate files.

I've tested the [workflow file](https://github.com/eunomia-bpf/WasmEdge/blob/master/.github/workflows/build-extensions.yml) (which adds tests for `wasm_bpf` to `build-extensions.yml`)on my local machine. The jobs run successfully.

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: utils/docker/Dockerfile.manylinux2014_x86_64 [hunk 1]
```diff
@@ -25,7 +25,8 @@ RUN cd && (yum check-update || true) && yum install -y xz openssl-devel rpm-buil
         https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.5/lld-16.0.5.src.tar.xz \
         https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.5/libunwind-16.0.5.src.tar.xz \
         https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.5/cmake-16.0.5.src.tar.xz \
-        https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.5/third-party-16.0.5.src.tar.xz && \
+        https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.5/third-party-16.0.5.src.tar.xz \
+        https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.5/clang-16.0.5.src.tar.xz && \
     sha256sum -c SHA256SUM && \
```
Comment (hydai) [inline utils/docker/Dockerfile.manylinux2014_x86_64:30]: The checksum file should also be updated and get the clang tarball verified.

**Review by hydai — CHANGES_REQUESTED** [review]

File: utils/docker/Dockerfile.manylinux2014_x86_64 (hunk 1 above)
Comment (Officeyutong) [reply utils/docker/Dockerfile.manylinux2014_x86_64:30]: Added now

**Review by hydai — CHANGES_REQUESTED** [review]
This PR only modifed manylinux2014 on x86_64. You should also update the aarch64 one.
https://github.com/WasmEdge/WasmEdge/blob/master/utils/docker/Dockerfile.manylinux2014_aarch64

**Comment by Officeyutong** [comment]
> This PR only modified manylinux2014 on x86_64. You should also update the aarch64 one. https://github.com/WasmEdge/WasmEdge/blob/master/utils/docker/Dockerfile.manylinux2014_aarch64 You can leave it alone if you don't have the plan to support arm architecture for the `wasm-bpf` plugin.

Currently I have no way to test `wasm_bpf` plugin on arm (though it should work theoretically), so `wasm_bpf` won't support arm now

**Review by hydai — APPROVED** [review]

**Comment by Officeyutong** [comment]
> Reference

Done

**Pull request merged**